components/portfolio/PortfolioTokenCard.tsx

```diff
@@ -0,0 +1,643 @@
+import {
+  faCheck,
+  faEdit,
+  faEllipsis,
+  faGasPump,
+  faHand,
+  faPlus,
+  faRefresh,
+} from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import {
+  EditListingModal,
+  EditListingStep,
+  extractMediaType,
+  TokenMedia,
+  useDynamicTokens,
+  useTokens,
+  useUserTokens,
+} from '@reservoir0x/reservoir-kit-ui'
+import { AcceptBid } from 'components/buttons'
+import BuyNow from 'components/buttons/BuyNow'
+import CancelListing from 'components/buttons/CancelListing'
+import List from 'components/buttons/List'
+import { spin } from 'components/common/LoadingSpinner'
+import {
+  Box,
+  Button,
+  Flex,
+  FormatCryptoCurrency,
+  Text,
+  Tooltip,
+} from 'components/primitives'
+import { Dropdown, DropdownMenuItem } from 'components/primitives/Dropdown'
+import { ToastContext } from 'context/ToastContextProvider'
+import { useMarketplaceChain } from 'hooks'
+import Link from 'next/link'
+import { UserToken } from 'pages/portfolio/[[...address]]'
+import {
+  Dispatch,
+  SetStateAction,
+  SyntheticEvent,
+  useContext,
+  useMemo,
+  useState,
+} from 'react'
+import { MutatorCallback } from 'swr'
+import { useMediaQuery } from 'react-responsive'
+import fetcher from 'utils/fetcher'
+import { formatNumber } from 'utils/numbers'
+import { DATE_REGEX, timeTill } from 'utils/till'
+import { zoneAddresses } from 'utils/zoneAddresses'
+import { Address } from 'wagmi'
+import Image from 'next/image'
+
+type PortfolioTokenCardProps = {
+  token: ReturnType<typeof useUserTokens>['data'][0]
+  address: Address
+  isOwner: boolean
+  rarityEnabled: boolean
+  tokenCount?: string
+  orderQuantity?: number
+  selectedItems: UserToken[]
+  setSelectedItems: Dispatch<SetStateAction<UserToken[]>>
+  mutate?: MutatorCallback
+  onMediaPlayed?: (
+    e: SyntheticEvent<HTMLAudioElement | HTMLVideoElement, Event>
+  ) => void
+}
+
+export default ({
+  token,
+  address,
+  isOwner,
+  rarityEnabled = true,
+  orderQuantity,
+  tokenCount,
+  selectedItems,
+  setSelectedItems,
+  mutate,
+  onMediaPlayed,
+}: PortfolioTokenCardProps) => {
+  const { addToast } = useContext(ToastContext)
+  const [isRefreshing, setIsRefreshing] = useState(false)
+
+  let dynamicToken = token as ReturnType<typeof useDynamicTokens>['data'][0]
+
+  const isSmallDevice = useMediaQuery({ maxWidth: 900 })
+
+  const mediaType = extractMediaType(dynamicToken?.token)
+  const showPreview =
+    mediaType === 'other' || mediaType === 'html' || mediaType === null
+  const { routePrefix, proxyApi } = useMarketplaceChain()
+
+  const orderZone = token?.ownership?.floorAsk?.rawData?.zone
+  // @ts-ignore
+  const orderKind = token?.ownership?.floorAsk?.kind
+
+  const isOracleOrder =
```

Comment:
We can replace this with the new property that was added in this PR: #231 
but we need to wait until that PR launches which will probably be soon

Reply:
PR was merged, we can updated this now.